File: services/currency_service.py
"""Сервис для работы с валютами"""
class CurrencyService:
    KRW_TO_RUB_RATE = 0.057279
    
    @staticmethod
    def get_exchange_rate() -> float:
        return CurrencyService.KRW_TO_RUB_RATE
    
    @staticmethod
    def convert_to_rub(price_won: float) -> int:
        if price_won is None:
            return None
        try:
            rate = CurrencyService.get_exchange_rate()
            return int(price_won * 10000 * rate / 1000)
        except (TypeError, ValueError) as e:
            print(f"Currency conversion error: {e}")
            return None
    
    @staticmethod
    def convert_to_won(price_rub: float) -> int:
        if price_rub is None:
            return None
        try:
            rate = CurrencyService.get_exchange_rate()
            return int(price_rub * 1000 / rate / 10000)
        except (TypeError, ValueError) as e:
            print(f"Currency conversion error: {e}")
            return None

File: static/js/services/currency-service.js
/**
 * Сервис конвертации валют - единый источник курса
 */
class CurrencyService {
    static getExchangeRate() {
        // Временная константа, позже можно получать через API
        return 0.057279;
    }
    
    static convertToRub(priceWon) {
        if (priceWon === null || priceWon === undefined) return null;
        try {
            const rate = this.getExchangeRate();
            return Math.round(priceWon * 10000 * rate / 1000);
        } catch (e) {
            console.error('Currency conversion error:', e);
            return null;
        }
    }
    
    static convertToWon(priceRub) {
        if (priceRub === null || priceRub === undefined) return null;
        try {
            const rate = this.getExchangeRate();
            return Math.round(priceRub * 1000 / rate / 10000);
        } catch (e) {
            console.error('Currency conversion error:', e);
            return null;
        }
    }

    static formatPriceRub(priceWon) {
        const rub = this.convertToRub(priceWon);
        return rub !== null ? `${rub.toLocaleString()} тыс.₽` : '—';
    }

    static formatPriceWon(priceWon) {
        return priceWon !== null ? `${(priceWon * 10000).toLocaleString()} ₩` : '—';
    }
}
